ui/app/src/model/datasource-api.ts

```diff
@@ -14,8 +14,10 @@
 import { Datasource, DatasourceSelector, GlobalDatasource } from '@perses-dev/core';
 import { DatasourceApi } from '@perses-dev/dashboards';
 import LRUCache from 'lru-cache';
-import { fetchDatasourceList, fetchGlobalDatasourceList } from './datasource-client';
+import { fetchDatasourceList } from './datasource-client';
+import { fetchGlobalDatasourceList } from './global-datasource-client';
 
+//TODO: Move to tanstack query for caching
```

Comment:
Definitely would love to see this using react-query, I briefly looked into it at one point and it wasn’t as easy as I hoped (my first attempt violated rules of hooks but there has to be a way)

Reply:
Same for me yeah I red the code and have been directly afraid of breaking something :fearful:  I think I will ask a bit of the time of @Nexucis at some point to do it together and be sure it follows what he wanted. I'm afraid he will have forgot as it's been a long time :laughing: